examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -786,16 +788,20 @@ def periodic_spline_transformer(period, n_splines=None, degree=3):
     "Splines + polynomial kernel",
     "Gradient Boosted Trees",
 ]
-for ax, pred, label in zip(axes, predictions, labels):
-    ax.scatter(y.iloc[test_0].values, pred, alpha=0.3, label=label)
-    ax.plot([0, 1], [0, 1], "--", label="Perfect model")
-    ax.set(
-        xlim=(0, 1),
-        ylim=(0, 1),
-        xlabel="True demand",
-        ylabel="Predicted demand",
-    )
-    ax.legend()
+plot_kinds = ["actual_vs_predicted", "residual_vs_predicted"]
+for axis_idx, kind in enumerate(plot_kinds):
+    for ax, pred, label in zip(axes[axis_idx], predictions, labels):
+        disp = PredictionErrorDisplay.from_predictions(
+            y_true=y.iloc[test_0],
+            y_pred=pred,
+            kind=kind,
+            scatter_kwargs={"alpha": 0.3},
+            ax=ax,
+        )
+        ax.set_xticks(np.linspace(0, 1, num=5))
+        ax.set_yticks(np.linspace(0, 1, num=5))
+        ax.set_aspect("equal", adjustable="box")
+        ax.legend(["Best model", label])
```

Comment:
The legend is a bit too large for the case "xxx + polynomial kernel". Increasing the figure size by putting the label on two lines (e.g. "xxx +\npolynomial kernel") would solve this issue.

Reply:
I am thinking that it would be enough to only add the legend on the first row and put the legend slightly outside the plot on the top (this would also solve the issue of the legend being too large).

A bit like this:

<img width="560" alt="image" src="https://user-images.githubusercontent.com/7454015/223767874-a67b371d-4a40-4a8d-87d8-b202be5eeb70.png">